lib/compat.php

```diff
@@ -319,16 +319,28 @@ function gutenberg_filter_oembed_result( $response, $handler, $request ) {
 
 	// External embeds.
 	if ( '/oembed/1.0/proxy' === $request->get_route() ) {
+		$local_oembed = false;
 		if ( is_wp_error( $response ) ) {
 			// It's possibly a local post, so lets try and retrieve it that way.
 			$post_id = url_to_postid( $_GET['url'] );
 			$data    = get_oembed_response_data( $post_id, apply_filters( 'oembed_default_width', 600 ) );
 
-			if ( ! $data ) {
-				// Not a local post, return the original error.
-				return $response;
+			if ( $data ) {
+				// It's a local post!
+				$local_oembed = true;
+				$response     = (object) $data;
+			}
+		}
+
+		if ( is_wp_error( $response ) || ! $local_oembed ) {
```

Comment:
If I'm reading correctly, couldn't this be an `else` to the above `if ( $data ) {`? The `is_wp_error( $response ) ||` seems redundant with the previous condition, also wouldn't need the `$local_oembed` variable:

```php
<?php

if ( is_wp_error( $response ) ) {
	// It's possibly a local post, so lets try and retrieve it that way.
	$post_id = url_to_postid( $_GET['url'] );
	$data    = get_oembed_response_data( $post_id, apply_filters( 'oembed_default_width', 600 ) );

	if ( $data ) {
		// It's a local post!
		$response = (object) $data;
	} else {
		global $wp_embed;
		$html = $wp_embed->shortcode( array(), $_GET['url'] );
		if ( $html ) {
			return array(
				'provider_name' => __( 'Embed Handler', 'gutenberg' ),
				'html'          => $html,
			);
		}
	}
}
```

Reply:
Good call, I think it was a left over from a previous iteration of the code: I wasn't able to reproduce any issues after changing it.